packages/lib/string-utils.js

```diff
@@ -301,20 +301,17 @@ function surroundKeywords(keywords, text, prefix, suffix, options = null) {
 		escapeHtml: false,
 	}, options);
 
-	if (!keywords.length) return text;
+	text = options.escapeHtml ? htmlentities(text) : text;
 
-	function escapeHtml(s) {
-		if (!options.escapeHtml) return s;
-		return htmlentities(s);
-	}
+	if (!keywords.length) return text;
 
 	let regexString = keywords
 		.map(k => {
 			if (k.type === 'regex') {
-				return escapeHtml(stringUtilsCommon.replaceRegexDiacritics(k.valueRegex));
+				return stringUtilsCommon.replaceRegexDiacritics(k.valueRegex);
 			} else {
 				const value = typeof k === 'string' ? k : k.value;
-				return escapeHtml(stringUtilsCommon.replaceRegexDiacritics(stringUtilsCommon.pregQuote(value)));
+				return stringUtilsCommon.replaceRegexDiacritics(stringUtilsCommon.pregQuote(value));
```

Comment:
Since we're no longer escaping `value`, isn't it possible again to do an XSS?

Unfortunately I see I didn't add a test when I fixed that security issue, so I'm not sure if this introducing a regression or not

Reply:
No, this will still prevent the XSS, because the text is escaped before it is added to the result list.
![image](https://user-images.githubusercontent.com/46966845/211627161-e60f11cd-af77-4c51-8246-016bf111db48.png)
the htmlentities method invalidates any html contained in the heading by replacing characters such as '<' with something like '&lt'.
As the html code is not valid the embedded js code is not executed.

Off course, I tested those payloads and they didn't work.